src/AppRoutes.tsx

```diff
@@ -91,7 +141,175 @@ const AsyncGames = Loadable({
+//   "title": "Dictionary",
+//   "author": "Typey Type",
+//   "category": "Typey Type",
+//   "tagline": "Typey Type’s dictionary is a version of the Plover di  ctionary with misstrokes removed for the top 10,000 words.",
+//   "subcategory": "",
+//   "link": "/support#typey-type-dictionary",
+//   "path": "/dictionaries/typey-type/typey-type.json"
+// }],
+  flashcardsMetWords: unknown, // TODO: type like {
+//   "the": {
+//     phrase: "the",
+//     stroke: "-T",
+//     rung: 0,
+//   },
+// },
+  globalLookupDictionary: LookupDictWithNamespacedDictsAndConfig,
+  globalLookupDictionaryLoaded: boolean,
+  lessonNotFound: boolean,
+  lessonsProgress: unknown,
+  flashcardsProgress: unknown,
+  flashcardsNextLesson: unknown, // TODO: type like {
+//   lastSeen: Date.now(), // Saturday, May 18, 2019 12:00:55 PM GMT+10:00
+//   linkTitle: "Loading…",
+//   linkText: "Study",
+//   link: process.env.PUBLIC_URL + "/lessons/drills/prefixes/flashcards"// + "?recommended=true&" + PARAMS.practiceParams
+// },
+// flashcardsCourseIndex: 0,
+  fullscreen: boolean,
+  globalUserSettings: GlobalUserSettings,
+// isPloverDictionaryLoaded: false,
+// isGlobalLookupDictionaryLoaded: false,
+  lookupTerm: string,
+  recommendationHistory: unknown // TODO: type like { currentStep: null },
+  personalDictionaries: ImportedPersonalDictionaries,
+  previousCompletedPhraseAsTyped: ActualTypedText,
+  repetitionsRemaining: number,
+  startTime: Date,
+  showStrokesInLesson: boolean,
+  targetStrokeCount: number,
+  timer: number,
+  topSpeedPersonalBest: number,
+  totalNumberOfMatchedWords: number,
+// numberOfMatchedChars: 0,
+// totalNumberOfMatchedChars: 0,
+  totalNumberOfNewWordsMet: number,
+  totalNumberOfLowExposuresSeen: number,
+  totalNumberOfRetainedWords: number,
+  totalNumberOfMistypedWords: number,
+  totalNumberOfHintedWords: number,
+  disableUserSettings: boolean,
+  metWords: MetWords,
+  revisionMode: boolean,
+  oldWordsGoalUnveiled: boolean,
+  newWordsGoalUnveiled: boolean,
+  userGoals: unknown // TODO type like {
+//   newWords: 15,
+//   oldWords: 50
+// },
+  userSettings: UserSettings,
+  lesson: Lesson,
+  lessonIndex: unknown, // TODO type like {
+//   "title": "Steno",
+//   "subtitle": "",
+//   "category": "Drills",
+//   "subcategory": "",
+//   "path": process.env.PUBLIC_URL + "/drills/steno/lesson.txt"
+// }],
+  recentLessons: { history: unknown },
+  recommendedNextLesson: unknown, // TODO type like {
+//   studyType: "practice",
+//   limitNumberOfWords: 50,
+//   repetitions: 1,
+//   linkTitle: "Top 10000 Project Gutenberg words",
+//   linkText: "Practice 150 words from Top 10000 Project Gutenberg wo  rds",
+// link: process.env.PUBLIC_URL + "/lessons/drills/top-10000-project-gutenberg-words/?recommended=true&" + PARAMS.practiceParams
+// },
+// revisionMaterial: [
+// ],
+  startingMetWordsToday: MetWords,
+  yourSeenWordCount: number,
+  yourMemorisedWordCount: number
+}
+type AppMethods = {
```

Comment:
type of the fields here are either:

* type of JS functions in `updateUserSetting`, e.g. `typeof fetchAndSetupGlobalDict`
* type of `App`'s method e.g. `typeof App.prototype.setupLesson`